src/components/20-molecules/datepicker/index.js

```diff
@@ -0,0 +1,369 @@
+  }
+
+  toggleDatepicker() {
+    if (!this.open) {
+      if (window.axaComponents.openDatepickerInstance && window.axaComponents.openDatepickerInstance !== this) {
+        window.axaComponents.openDatepickerInstance.open = false;
+      }
+      this.open = true;
+      window.axaComponents.openDatepickerInstance = this;
+    } else {
+      this.open = false;
+      window.axaComponents.openDatepickerInstance = null;
+    }
+  }
+
+  isValidDate(date) {
+    let out = false;
+    try {
+      const parsedDate = new Date(Date.parse(date));
+      // eslint-disable-next-line no-restricted-properties
+      const isValid = parsedDate instanceof Date && !window.isNaN(parsedDate);
+      const isValidDateLocalized = parseLocalisedDateIfValid(this.locale, date);
+      if ((isValid && isValidDateLocalized) || (!isValid && isValidDateLocalized)) {
+        out = isValidDateLocalized;
+      }
+    } catch (e) {
+      out = false;
+    }
+    return out;
+  }
+
+  render() {
+    return html`
+      <article class="m-datepicker">
+        ${this.inputField &&
+          html`
+            <div class="m-datepicker__input-wrap">
+              <input
+                @keyup="${this.handleInputChange}"
+                class="m-datepicker__input js-datepicker__input"
+                type="text"
+                placeholder="Please select a date"
+                value="${this.outputDate}"
+              />
+              <button class="m-datepicker__input-button" @click="${this.handleInputButtonClick}">
+                <span class="m-datepicker__input-icon">${iconDatepicker}</span>
+              </button>
+            </div>
+          `}
+        <div class="m-datepicker__wrap">
+          <div class="m-datepicker__article">
+            <div class="m-datepicker__dropdown-wrap">
+              <axa-dropdown
+                class="m-datepicker__dropdown m-datepicker__dropdown-month js-datepicker__dropdown-month"
+                max-height
+                items="${JSON.stringify(this.monthItems)}"
+                title="Month"
+              >
+              </axa-dropdown>
+
+              <axa-dropdown
+                class="m-datepicker__dropdown m-datepicker__dropdown-year js-datepicker__dropdown-year"
+                max-height
+                items="${JSON.stringify(this.yearItems)}"
+                title="Year"
+              >
+              </axa-dropdown>
+            </div>
+
+            <div class="m-datepicker__weekdays">
+              ${this.weekdays &&
+                this.weekdays.map(
+                  day =>
+                    html`
+                      <div class="m-datepicker__weekdays-day">${day}</div>
+                    `
+                )}
+            </div>
+
+            <div class="m-datepicker__calendar js-datepicker__calendar">
+              ${this.cells &&
+                this.cells.map(
+                  (cell, index) =>
+                    html`
+                      <button
```

Comment:
carefull again about default type being submit and without a form is invalid

Reply:
done